programs/survival/enchanting_table_bottling.sc

```diff
@@ -0,0 +1,18 @@
+global_xp_cost = 8;
+
+__on_player_right_clicks_block(player, item, hand, block, f, h) -> (
+	if(
+		player~'gamemode_id'%2==0 && //survival or adventure mode
+		item:0 == 'glass_bottle' &&
+		item:2 == null && //doesn't have nbt
+		block == 'enchanting_table' &&
+		player~'sneaking' && //they didn't open the UI
+		player~'xp' >= global_xp_cost, //they ahve enough xp
```

Comment:
Suggestion for checking the sneaking condition first (or after game mode) as it seems both a very cheap check and one that'll easily cancel the rest of checks in many cases.

Reply:
I thought that since the block and item had already been queried and saved into a variable by the event handler, those checks would be cheaper than the sneak one. Ut was just a guess, tho.